CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/gui/VisualGraphOpener.java

```diff
@@ -176,12 +178,45 @@ protected Void doInBackground() throws Exception {
             if (graph == null) {
                 try {
                     final long t0 = System.currentTimeMillis();
+                    LOGGER.log(Level.INFO, String.format("MMDEBUG Attempting to open %s", graphFile.toString()));
                     graph = new GraphJsonReader().readGraphZip(graphFile, new HandleIoProgress(String.format("Reading %s...", graphFile.getName())));
                     time = System.currentTimeMillis() - t0;
-                } catch (GraphParseException | IOException | RuntimeException ex) {
+                } catch (GraphParseException | IOException | RuntimeException ex) {  
                     gex = ex;
                 }
 
+                try {
+                    if (gex != null) {
+                        // An exception was thrown trying to read specified star file. The most likely reason for this is
+                        // a corrupt star file. Check to see if there was a 'backup' star file generated before the star file
+                        // was writtien - if so, attmept to load this.
+                        String backupFilename = graphFile.toString().concat(BACKUP_EXTENSION);
+                        File backupFile = new File(backupFilename);
```

Comment:
Both these can be made final

Reply:
addressed - also looked for others in file.